src/opt/Optimizer/Optimizer.hs

```diff
@@ -128,3 +129,30 @@ dropBorrowBlocks = extend dropBorrowBlock
            ,decls = [([VarNoType name], target)]
            ,body}
       dropBorrowBlock e = e
+
+forwardGeneral = extend forwardGeneral'
+  where
+    forwardGeneral' e@(Forward{forwardExpr=MessageSend{}}) = e
```

Comment:
this case and the one below can be subsumed by the last pattern, `forwardGeneral' e = e`